### DISCUSSION TITLE
How to test HttpHandler

### DISCUSSION THREAD

**Discussion by Al4ric** [giraffe-fsharp/Giraffe#510] [Q&A]
I wanted to use Giraffe with more TDD approach. But I'm not sure what is the best way to test HttpHandlers. Are there any good examples?

**Comment by TheAngryByrd** [comment]
Not an immediate example in F#, the [Microsoft TestHost](https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-6.0#aspnet-core-integration-tests) would be the basis for the answer.

**Reply by Al4ric** [reply]
Yes those are good for integration tests (also very valuable), but what about some unit tests?

**Comment by 64J0** [comment]
Check those tests: https://github.com/giraffe-fsharp/Giraffe/blob/master/tests/Giraffe.Tests/HttpHandlerTests.fs
